src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -399,15 +393,7 @@ private MgmtExtensions EnsureExtensions(Type armCoreType, RequestPath contextual
             return new MgmtExtensions(operations, armCoreType, contextualPath);
         }
 
-        private ArmClientExtensions? _armClientExtensions;
-        private ArmClientExtensions EnsureArmClientExtensions()
-        {
-            if (_armClientExtensions != null)
-                return _armClientExtensions;
-
-            _armClientExtensions = new ArmClientExtensions(GetChildOperations(RequestPath.Tenant));
-            return _armClientExtensions;
-        }
+        private ArmClientExtensions EnsureArmClientExtensions() => new ArmClientExtensions(GetChildOperations(RequestPath.Tenant));
```

Comment:
The previous code is more like a singleton, but new code is not, is this the expected change?

Reply:
All these instances work like a singleton because they only initialize once. Previously we are mixing two patterns:
First:
```csharp
private Something _property;
public Something Property => EnsureProperty();

private Something EnsureProperty(){
if (_property != null)
return _property;

_property = new Something();
return _property;
}
```
Second:
```csharp
private Something _property;
public Something Property => _property ??= EnsureProperty();

private Something EnsureProperty(){
return new Something();
}
```
These two are equivalent therefore I am unifying our code to use the simpler one (the second)